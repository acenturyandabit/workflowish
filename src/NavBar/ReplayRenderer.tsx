import * as React from 'react';
import Dialog from '@mui/material/Dialog';
import { Button, DialogActions, DialogContent, DialogTitle } from '@mui/material';
import { BaseDeltaType } from '~CoreDataLake';

export const ReplayRendererNavbarAndDialog = (props: {
    replayBuffer: BaseDeltaType[]
}) => {
    const [open, setOpen] = React.useState<boolean>(false);
    return <li>
        <Dialog open={open}
            onClose={() => setOpen(false)}
            fullWidth
            maxWidth="md">
            <DialogTitle>Undo</DialogTitle>
            <DialogContent className='viewContainer' style={{ color: "white" }} dividers={true}>
                {props.replayBuffer.map((delta, idx) => {
                    return <p key={idx}>{delta.key}</p>
                })}
            </DialogContent>
            <DialogActions>
                <Button onClick={() => setOpen(false)}>Close</Button>
            </DialogActions>
        </Dialog>
        <a onClick={() => { setOpen(true); }}>Show undo buffer</a>
    </li>
}