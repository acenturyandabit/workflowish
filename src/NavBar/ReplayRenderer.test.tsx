import * as renderer from 'react-test-renderer';
import * as React from "react";
import { it, expect } from '@jest/globals';
import { ReplayRendererNavbarAndDialog } from './ReplayRenderer';
import { BaseDeltaType } from '~CoreDataLake';

it("Doesn't render replay buffer if not shown", () => {
    const hugeReplayBuffer: BaseDeltaType[] = Array(100).fill({
        key: "dummy"
    });
    const component = renderer.create(
        <ReplayRendererNavbarAndDialog replayBuffer={hugeReplayBuffer}></ReplayRendererNavbarAndDialog>
    )
    const item = component.toJSON();
    expect(item).toMatchSnapshot();
})