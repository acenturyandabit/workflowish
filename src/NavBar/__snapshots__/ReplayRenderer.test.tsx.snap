// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Doesn't render replay buffer if not shown 1`] = `
<li>
  <a
    onClick={[Function]}
  >
    Show undo buffer
  </a>
</li>
`;
